log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeployed/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+        "type": "linear",
+        "domain": {"data": "geoData","field": "zipped.v"},
+        "domainMin": 0.0,
+        "zero": false,
+        "range":  config.colorScale
+    };
+
+    var scales =  [cScale];
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+    this.spec.predicates = predicates;
+    this.spec.legends = legends;
+
+};
+
+map.prototype.draw = function(div, callbacks) {
+    var viewUpdateFunction = (function(chart) {
+        this.view = chart({el:div}).renderer(this.config.renderer).update();
+
+        if (callbacks != null) {
+            for (var i = 0; i<callbacks.length; i++) {
+                this.view.on(callbacks[i].type, callbacks[i].callback);
+            }
+        }
+
+    }).bind(this);
+
+    vg.parse.spec(this.spec, viewUpdateFunction);
+};
+
+map.prototype.insert = function(data) {
+
+    var xAxis = this.metadata.names[this.config.x];
+    var yAxis = this.metadata.names[this.config.y];
+    var color = this.metadata.names[this.config.color];
+    var mapType = this.config.mapType;
+    var geoInfoJson = this.config.geoInfoJson;
+
+    for (i = 0; i < data.length; i++) {
+        for (var key in data[i]) {
+            if(key == xAxis){
+                if (data[i].hasOwnProperty(key)) {
+                    data[i].unitName = data[i][key];
+                    data[i][key] = getMapCode(data[i][key], mapType,geoInfoJson);
+                    break;
+                }
+            }
+        }
+    };
+
+    for (i = 0; i < data.length; i++) {
+        var isValueMatched = false;
+        this.view.data(this.config.title).update(function(d) {
+                return d[xAxis] == data[i][xAxis]; },
+            yAxis,
+            function(d) {
+                isValueMatched = true;
+                return data[i][yAxis];
+            });
+
+        this.view.data(this.config.title).update(function(d) {
+                return d[xAxis] == data[i][xAxis]; },
+            color,
+            function(d) {
+                isValueMatched = true;
+                return data[i][color];
+            });
+
+
+        if(!isValueMatched){
+            this.view.data(this.config.title).insert([data[i]]);
+        }
+    }
+    this.view.update();
+
+};
+
+function getTopoJson(config, metadata){
+
+    var width = config.width;
+    var height = config.height;
+    var scale;
+    var mapType = config.charts[0].mapType;
+    var projection = "mercator";
+
+    if(mapType == "usa"){
+        width = config.width - 160;
+        height = config.height - 130;
+        scale = config.height + 270;
+        projection = "albersUsa";
+    }else if(mapType == "europe"){
+        width = ((config.width/2)+ 100)/2;
+        height = config.height + 150;
+        scale = config.height + 50;
+    }else{
+        scale = (config.width/640)*120;
```

Comment:
Hi Sinthuja,

We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD